src/main/java/com/exasol/adapter/commontests/scalarfunction/ScalarFunctionsTestBase.java

```diff
@@ -143,550 +129,449 @@ private void runOnExasol(final ExasolExecutable exasolExecutable) {
      * @return {@code true} if test is disabled by dialect
      */
     private boolean isTestDisabledFor(final String function) {
-        final Set<String> dialectSpecificExcludes = getDialectSpecificExcludes();
+        final Set<String> dialectSpecificExcludes = getCachedTestSetup().getDialectSpecificExcludes();
         if (dialectSpecificExcludes.contains(function.toLowerCase())) {
             LOGGER.log(Level.FINE, "Skipping test for {0} since it was disabled for this dialect.", function);
             return true;
         }
         return false;
     }
 
-    @FunctionalInterface
-    private interface ExasolExecutable {
-        void runOnExasol(Statement statement) throws SQLException;
+    private VirtualSchemaTestSetup buildVirtualSchemaTableWithColumnOfExasolType(final DataType exasolType,
+            final Object valueForSingleRow) {
+        final String booleanType = getCachedTestSetup().getDataTypeThatThisAdapterMapsTo(exasolType);
+        final VirtualSchemaTestSetupRequest request = new VirtualSchemaTestSetupRequest(TableRequest.builder("MY_TABLE")
+                .column(MY_COLUMN, booleanType).row(List.of(valueForSingleRow)).build());
+        return getTestSetup().getVirtualSchemaTestSetupProvider().createSingleTableVirtualSchemaTestSetup(request);
```

Comment:
perhaps we can provided this funtionality directly from the `TestSetup`?
```suggestion
        return getTestSetup().createSingleTableVirtualSchemaTestSetup(request);
```